### ISSUE TITLE
NLog stops working when "Output path" is not bin/debug by default

### ISSUE THREAD

**Issue by sekulicb** [NLog/NLog#3038]
NLog version: 4.5.1

Platform: .Net 4.5 ( Console application with 4.6.2 target framework) 

Hi. I noticed one big problem that occurs with projects where "Platform target" is set to x64 in Visual Studio 2017. I downloaded latest nuget package in one of the apps that I'm working at for a client. Its WPF application. Now the config could not be much simpler, and it looks like this:

```
<targets>
    <target name="logfile" xsi:type="File" fileName="D:/logfiles/LogFile.txt" />
  </targets>

  <rules>
    <logger name="*" minlevel="Info" writeTo="logfile" />
  </rules> 

```
But the logs are never there. I doubled checked if I have permission to write to this file and everything is ok for those external rules. I also created a small console app with the same target framework, just to verify. So when I run it in the Main method of the console app:

 class Program
    {
        private static Logger logger = LogManager.GetCurrentClassLogger();

        static void Main(string[] args)
        {           
            logger.Trace("Sample trace message");
            logger.Debug("Sample debug message");
            logger.Info("Sample informational message");
            logger.Warn("Sample warning message");
            logger.Error("Sample error message");
            logger.Fatal("Sample fatal error message");
            Console.WriteLine("done!");
            Console.ReadLine();
        }
    }

I can see that logs are there and everything works fine. So it took me a while to compare things with my WPF project, and the moment I set that property in console app, logs are simply not written anymore.

The problem is that the app is not mine, internally it uses some VB code and its tied to some bigger platform projects, so I cannot just change this value (I assume client had its reasons for setting this up). 

What should I do ? Is there a solution for this?

**Comment by snakefoot** [comment]
I have used NLog with x64-applications without any issues, so I think your issue is the application cannot find the `NLog.config`-file. Good start is to verify that the `NLog.config` is copied correctly to the exe-output-path. Next step is to add `throwConfigExceptions="true"` and activate NLog InternalLogger: https://github.com/NLog/NLog/wiki/Internal-Logging

**Comment by sekulicb** [comment]
Good point. Looking at the WPF project, exe path is set to virtual directory (there is a .bat that executes **subst** command), so multiple projects point to that path and can share dependencies. For example instead of default _bin/debug_ its pointing to X drive. But either way, it doesn't matter where build output is, I can see NLog.config in the same directory as .exe of the application. Can the problem be with non-default build path?

I have enabled throwExceptions="true" and throwConfigExceptions="true" but I don't see those logs as well. Here is the full Nlog.config: 

`<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://www.nlog-project.org/schemas/NLog.xsd NLog.xsd"
      autoReload="true"      
      throwExceptions="true"
      throwConfigExceptions="true"
      internalLogLevel="Trace" internalLogFile="c:\temp\nlog-internal.log">

  <targets>
    <target name="logfile" xsi:type="File" fileName="D:/logfiles/LogFile.txt" />
  </targets>

  <rules>
    <logger name="*" minlevel="Info" writeTo="logfile" />
  </rules> 
</nlog>
`

Thanks

**Comment by sekulicb** [comment]
I did. Same thing...noting happens. I just can't believe that if you change your output path, logger stops working. Is there any workaround this ?

Can you confirm this on your end. Just create simple console app in VS, download nuget and change output path. I think this is bug, or at least there should be a way to reference exe folder if not by default.

Thanks

**Comment by snakefoot** [comment]
You are welcome to attach a sln-project that reproduces the issue. NLog is very eager about loading its configuration. Either from app.config or nlog.config in the basedirectory. So the standard problem is people forgetting to configure `Copy Always` in the File-properties. See also https://github.com/NLog/NLog/wiki/Tutorial#configure-nlog-targets-for-output

**Comment by sekulicb** [comment]
I configured everything, and checked. The problem is that you _cannot use NLog.config_ if your output path is not equal to bin/debug. 

Anyway you input solved my problem, I followed your link and managed to programmatically configure NLog. This should noted somewhere. If anybody in the future has the same problem, use [Configuration API](https://github.com/NLog/NLog/wiki/Configuration-API) and that's it.  

This issue can be closed. Thanks again @snakefoot
